jitsi-utils-kotlin/src/main/kotlin/org/jitsi/utils/stats/TimeBasedSlidingWindow.kt

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright @ 2018 - present 8x8, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.jitsi.utils.stats
+
+import java.time.Clock
+import java.time.Duration
+import java.time.Instant
+
+@ExperimentalStdlibApi
+open class TimeBasedSlidingWindow<T : Any>(
+    private val windowSize: Duration,
+    private val clock: Clock = Clock.systemUTC(),
+    private val evicted: (TimeEntry<T>) -> Unit = {}
```

Comment:
I'd list 'evicted' before 'clock' in case you want to add a `@JvmOverloads` here